frontend-library/src/utils/Http.service.ts

```diff
@@ -8,7 +8,8 @@ import axiosRetry from 'axios-retry';
 export class HttpService {
   private readonly axiosInstance: AxiosInstance;
   private isTokenRefreshing: boolean = false;
-  private userToken: string;
+  public userToken: string;
+  private onUserTokenChangeCallback: (token: string) => void;
```

Comment:
Consider using common interface for subscriptions https://developer.mozilla.org/en-US/docs/Web/API/EventTarget